internal/backends/qt/qt_accessible.rs

```diff
@@ -0,0 +1,699 @@
+use i_slint_core::properties::{PropertyChangeHandler, PropertyTracker};
+use i_slint_core::SharedVector;
+
+use cpp::*;
+use pin_project::pin_project;
+use qttypes::QString;
+
+use alloc::boxed::Box;
+use core::ffi::c_void;
+use std::pin::Pin;
+
+pub struct AccessibleItemPropertiesTracker {
+    accessible_item: *mut c_void,
+}
+
+// KEEP IN SYNC WITH CONSTANTS IN C++
+const NAME: u32 = QAccessible_Text_Name;
+const DESCRIPTION: u32 = QAccessible_Text_Description;
+const VALUE: u32 = QAccessible_Text_Value;
+const CHECKED: u32 = QAccessible_Text_UserText as u32;
+const VALUE_MINIMUM: u32 = CHECKED + 1;
+const VALUE_MAXIMUM: u32 = VALUE_MINIMUM + 1;
+const VALUE_STEP: u32 = VALUE_MAXIMUM + 1;
+
+impl PropertyChangeHandler for AccessibleItemPropertiesTracker {
+    fn notify(&self) {
+        let accessible_item = self.accessible_item;
+        cpp!(unsafe [accessible_item as "Slint_accessible_item*"] {
+            QTimer::singleShot(0, [accessible_item]() {
+                auto data = accessible_item->data();
+                rust!(AccessibleItemPropertiesTracker_rearm [data: Pin<&SlintAccessibleItemData> as "void*"] {
+                    data.arm_state_tracker();
+                });
+
+                auto obj = accessible_item->object();
+                QAccessible::State s = {};
+                s.checked = true; // Mark checked as changed!
+                auto event = QAccessibleStateChangeEvent(obj, s);
+                QAccessible::updateAccessibility(&event);
+            });
+        });
+    }
+}
+
+pub struct ValuePropertyTracker {
+    accessible_item: *mut c_void,
+}
+
+impl PropertyChangeHandler for ValuePropertyTracker {
+    fn notify(&self) {
+        let accessible_item = self.accessible_item;
+        cpp!(unsafe [accessible_item as "Slint_accessible_item*"] {
+            QTimer::singleShot(0, [accessible_item]() {
+                auto data = accessible_item->data();
+                rust!(ValuePropertyTracker_rearm [data: Pin<&SlintAccessibleItemData> as "void*"] {
+                    data.arm_value_tracker();
+                });
+
+                qDebug() << "Value Change signalled!";
+                auto obj = accessible_item->object();
+                auto event = QAccessibleValueChangeEvent(obj, accessible_item->currentValue());
+                QAccessible::updateAccessibility(&event);
+            });
+        });
+    }
+}
+
+pub struct LabelPropertyTracker {
+    accessible_item: *mut c_void,
+}
+
+impl PropertyChangeHandler for LabelPropertyTracker {
+    fn notify(&self) {
+        let accessible_item = self.accessible_item;
+        cpp!(unsafe [accessible_item as "Slint_accessible_item*"] {
+            QTimer::singleShot(0, [accessible_item]() {
+                auto data = accessible_item->data();
+                rust!(LabelPropertyTracker_rearm [data: Pin<&SlintAccessibleItemData> as "void*"] {
+                    data.arm_label_tracker();
+                });
```

Comment:
As a thought for the future: Instead of manually arming all the trackers, it would be nice if that were to happen automatically. After calling `QAccessible::updateAccessibility` I suppose Qt will call the various getter functions to query description, etc.. If those implementations could use a new `evaluate_as_dependency_root_append` function of PropertyTracker that doesn't clear the existing dependency nodes, then the manual book-keeping of what properties are dependencies goes away.

Reply:
> Something that would just stay armed would be nice

Not possible with the current implementation of the lazy property.
Without evaluating the binding we can't get the dependencies.


But the point is that we should only "arm" it if it was queried before.  Like you don't need to send notification of changes if nobody queries it.  Anyway, i think it's out of the scope of this patch.

Reply:
I'll close this: I agree that this is out of scope for this PR.